pkg/virt-launcher/virtwrap/agent-poller/agent_poller_test.go

```diff
@@ -244,6 +244,269 @@ var _ = Describe("Qemu agent poller", func() {
 			Expect(commandExecutions).To(Equal(expectedExecutions))
 		})
 	})
+
+	Context("Load averaging functionality", func() {
+		var fakeLoad api.Load
+
+		BeforeEach(func() {
+			fakeLoad = api.Load{
+				Load1m:  1.23,
+				Load5m:  2.45,
+				Load15m: 3.67,
+			}
+		})
+
+		Context("convertToLoad", func() {
+			It("should convert libvirt DomainGuestInfo load to api.Load", func() {
+				guestInfo := &libvirt.DomainGuestInfo{
+					Load: &libvirt.DomainGuestInfoLoad{
+						Load1M:  1.23,
+						Load5M:  2.45,
+						Load15M: 3.67,
+					},
+				}
+
+				load := convertToLoad(guestInfo)
+
+				Expect(load.Load1m).To(Equal(1.23))
+				Expect(load.Load5m).To(Equal(2.45))
+				Expect(load.Load15m).To(Equal(3.67))
+			})
+
+			It("should return empty Load when guestInfo.Load is nil", func() {
+				guestInfo := &libvirt.DomainGuestInfo{
+					Load: nil,
+				}
+
+				load := convertToLoad(guestInfo)
+
+				Expect(load.Load1m).To(Equal(0.0))
+				Expect(load.Load5m).To(Equal(0.0))
+				Expect(load.Load15m).To(Equal(0.0))
+			})
+
+			It("should handle zero load values", func() {
+				guestInfo := &libvirt.DomainGuestInfo{
+					Load: &libvirt.DomainGuestInfoLoad{
+						Load1M:  0.0,
+						Load5M:  0.0,
+						Load15M: 0.0,
+					},
+				}
+
+				load := convertToLoad(guestInfo)
+
+				Expect(load.Load1m).To(Equal(0.0))
+				Expect(load.Load5m).To(Equal(0.0))
+				Expect(load.Load15m).To(Equal(0.0))
+			})
+
+			It("should handle high load values", func() {
+				guestInfo := &libvirt.DomainGuestInfo{
+					Load: &libvirt.DomainGuestInfoLoad{
+						Load1M:  100.50,
+						Load5M:  200.75,
+						Load15M: 300.99,
+					},
+				}
+
+				load := convertToLoad(guestInfo)
+
+				Expect(load.Load1m).To(Equal(100.50))
+				Expect(load.Load5m).To(Equal(200.75))
+				Expect(load.Load15m).To(Equal(300.99))
+			})
+		})
+
+		Context("Load storage in AsyncAgentStore", func() {
+			It("should store and retrieve load data", func() {
+				agentStore.Store(libvirt.DOMAIN_GUEST_INFO_LOAD, fakeLoad)
+
+				sysInfo := agentStore.GetSysInfo()
+				Expect(sysInfo.Load.Load1m).To(Equal(fakeLoad.Load1m))
+				Expect(sysInfo.Load.Load5m).To(Equal(fakeLoad.Load5m))
+				Expect(sysInfo.Load.Load15m).To(Equal(fakeLoad.Load15m))
+			})
+
+			It("should fire an event for new load data", func() {
+				agentStore.Store(libvirt.DOMAIN_GUEST_INFO_LOAD, fakeLoad)
+
+				Expect(agentStore.AgentUpdated).To(Receive(Equal(AgentUpdatedEvent{
+					DomainInfo: api.DomainGuestInfo{
```

Comment:
Assert that received object contains the expected `fakeLoad`?

Reply:
Did you push that change?

Reply:
Yes line 317